core/modules/system/system.install

```diff
@@ -2650,6 +2650,97 @@ function system_update_1045() {
 }
 
 /**
+ * Save the current Bartik theme palette as a custom palette.
+ */
+function system_update_1046() {
+  global $theme;
+
+  // Only bother if the current theme is Bartik.
+  if ($theme == 'bartik') {
```

Comment:
when update.php is running the active theme is `seven`. We need to check the default theme instead.